cinema/tests/test_cinema_hall_api.py

```diff
@@ -0,0 +1,116 @@
+from django.test import TestCase
+
+from rest_framework.test import APIClient
+from rest_framework import status, generics
+
+from cinema.serializers import CinemaHallSerializer
+from cinema.models import CinemaHall
+from cinema.views import CinemaHallViewSet
+
+
+class CinemaHallApiTests(TestCase):
+    def setUp(self):
+        self.client = APIClient()
+        CinemaHall.objects.create(
+            id=1,
+            name="Blue",
+            rows=15,
+            seats_in_row=20,
+        )
+        CinemaHall.objects.create(
+            id=2,
+            name="VIP",
+            rows=6,
+            seats_in_row=8,
+        )
+
+    def test_cinema_hall_is_subclass_generic_api_view(self):
+        self.assertEqual(
+            issubclass(CinemaHallViewSet, generics.GenericAPIView), True
+        )
+
+    def test_get_cinema_halls(self):
+        response = self.client.get("/api/cinema/cinema_halls/")
+        serializer = CinemaHallSerializer(CinemaHall.objects.all(), many=True)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertEqual(response.data, serializer.data)
+
+    def test_post_cinema_halls(self):
+        response = self.client.post(
+            "/api/cinema/cinema_halls/",
+            {
+                "id": 3,
```

Comment:
This is also unneccessary, Post endpoints must be without id, it should be impossible to provide one